autoprof/fit/iterative.py

```diff
@@ -80,7 +111,12 @@ def step(self):
 
         self.iteration += 1
         
-    def fit(self):
+    def fit(self) -> None:
```

Comment:
`fit` returns `self` not `None`

Reply:
Same as before by returns `self` I should have said it returns a `BaseOptimizer` object